lua/wire/stools/expression2.lua

```diff
@@ -55,20 +72,135 @@ if SERVER then
 		end
 	end
 
-	util.AddNetworkString("WireExpression2_OpenEditor")
-	function TOOL:RightClick(trace)
-		if trace.Entity:IsPlayer() then return false end
+	-- Steam ID bypass for trusted users (anyone who already has SV Lua anyway pretty much)
+	local bypassList = {}
+	concommand.Add("wire_expression2_viewrequest_whitelist", function(plr, cmdName, args, argStr)
+		if IsValid(plr) then
+			print("This command can only be used on the server")
+			return
+		end
 
-		local player = self:GetOwner()
+		bypassList = {}
+		for _, v in pairs(args) do
+			v = tostring(v) -- (tostring isn't "needed", just a sanity check)
+			if v and util.SteamIDTo64(v) ~= "0" then -- Validate that this argument is, indeed, a steam id
+				bypassList[v] = true
+			end
+		end
+	end, nil, "List of SteamIDs that are able to bypass the view request system")
+
+	-- Simple serverside only local table for storing view requests to make handling them not spaghetti code
+	local viewRequests = {}
+
+	hook.Add("Tick", "WireExpression2_InvalidateRequests", function()
```

Comment:
I would rewrite this to not use a hook at all. 
We don't need to actively purge these requests from memory, but can do them instead when requests are being generated, no?

Reply:
The iteration would need to still be done (otherwise a possible memory leak), but doing it when new requests are generated is far better, yeah.  

Easy fix but difficult to push right now as I have other unfinished code in there for the collaboration system I talked about above.

Reply:
Actually @Python1320, removing this hook would then require using timers for notifying the asking client when their request expires, and I doubt creating an unknown quantity of timers would be more efficient than just enumerating the view requests each tick.  

Could add a counter with modulo to only check every few ticks or so though.

Reply:
Make the Think hook add when there's an entry in the table and remove it when the table is empty.